### PR TITLE
Add SASL_PLAINTEXT/SASL_SSL security protocols

### PR DESCRIPTION
**Pull request by millin**
Hi everyone! 
This is Alexander Millin from Citymobil.

I added the ability to connect to Kafka with SASL_PLAINTEXT or SASL_SSL security protocols.
I tested it in our environment with SASL_SSL and it seems to work fine.

Please kindly review.
If necessary, you are free to modify this or ask me to fix something, as I have never written on Scala before :smile:

### CODE REVIEW

**Comment by morazow** [comment]
Hello @millin,

This is great! Thanks for the pull request! I am going to review it soon.

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 1]
```diff
@@ -146,9 +148,15 @@ class KafkaConsumerProperties(private val properties: Map[String, String]) exten
   final def getMaxRecordsPerRun(): Int =
     get(MAX_RECORDS_PER_RUN.userPropertyName).fold(MAX_RECORDS_PER_RUN.defaultValue)(_.toInt)
 
-  /** Checks if the {@code SSL_ENABLED} property is set. */
+  /** Checks if {@code SECURITY_PROTOCOL} is using {@code SSL channel}. */
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:151]:
```suggestion
  /** Checks if {@code SECURITY_PROTOCOL} is using {@code SSL} channel. */
```

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 2]
```diff
@@ -146,9 +148,15 @@ class KafkaConsumerProperties(private val properties: Map[String, String]) exten
...
   final def isSSLEnabled(): Boolean =
-    isEnabled(SSL_ENABLED)
+    List(SecurityProtocol.SSL, SecurityProtocol.SASL_SSL)
+      .contains(SecurityProtocol.valueOf(getSecurityProtocol()))
+
+  /** Checks if {@code SECURITY_PROTOCOL} is using {@code SASL authentication}. */
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:156]:
```suggestion
  /** Checks if {@code SECURITY_PROTOCOL} is using {@code SASL} authentication. */
```

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 3]
```diff
@@ -323,6 +354,8 @@ object KafkaConsumerProperties extends CommonProperties {
   /**
    * An optional property key name to set SSL secure connections to
    * Kafka cluster.
+   *
+   * @deprecated("Use SECURITY_PROTOCOL=\"SSL\" instead", "1.2.2")
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:358]: :+1:

File: src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala [hunk 4]
```diff
@@ -151,7 +165,7 @@ class KafkaConsumerPropertiesTest extends AnyFunSuite with BeforeAndAfterEach wi
     val thrown = intercept[IllegalArgumentException] {
       KafkaConsumerProperties(properties).getProperties()
     }
-    assert(thrown.getMessage() === errorMessage("SCHEMA_REGISTRY_URL"))
+    assert(thrown.getMessage === errorMessage("SCHEMA_REGISTRY_URL"))
```
Comment (morazow) [inline src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala:168]: Please revert.

Comment (morazow) [reply src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala:168]: We adopted to keep parantheses for method even if they do not accept parameters, this way it is easier to distinguish between variables and methods with no parameters.

File: src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala [hunk 5]
```diff
@@ -373,8 +398,8 @@ class KafkaConsumerPropertiesTest extends AnyFunSuite with BeforeAndAfterEach wi
     val metadata = mock[ExaMetadata]
     val connectionInformation = mock[ExaConnectionInformation]
     when(metadata.getConnection("MY_CONNECTION")).thenReturn(connectionInformation)
-    when(connectionInformation.getUser()).thenReturn("")
-    when(connectionInformation.getPassword())
+    when(connectionInformation.getUser).thenReturn("")
```
Comment (morazow) [inline src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala:401]: Please revert here and other places. 

I think you had formatting rule to remove them.

**Review by morazow — CHANGES_REQUESTED** [review]
I have added some suggestions.

Could you please also add short example to user guide with SASL protocol?

Thanks a lot! We really appreciate your contributions!

File: src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala (hunk 4 above)
Comment (millin) [reply src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala:168]: Ok

Comment (millin) [reply src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala]: I reverted all removed parentheses, but found that usage of methods with/without parentheses are currently mixed:

https://github.com/exasol/kafka-connector-extension/blob/0cfa53936c32d286af57d2bbcd45d7725b361a1e/src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala#L35
 
https://github.com/exasol/kafka-connector-extension/blob/0cfa53936c32d286af57d2bbcd45d7725b361a1e/src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala#L154

**Comment by millin** [comment]
I reverted all removed parentheses, but found that usage of methods with/without parentheses are currently mixed:

https://github.com/exasol/kafka-connector-extension/blob/0cfa53936c32d286af57d2bbcd45d7725b361a1e/src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala#L35
 
https://github.com/exasol/kafka-connector-extension/blob/0cfa53936c32d286af57d2bbcd45d7725b361a1e/src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerPropertiesTest.scala#L154

**Comment by morazow** [comment]
Thanks @millin.
Yes, there are still the cases from early development, I would appreciate if you could also add parentheses to these methods.

**Comment by morazow** [comment]
Hello @millin,

I have added changes files and run formatter (unfortunately CI task do not run on remote forks).

Could you please add short import example using sasl protocol to user guide? Another subsection with sasl example would be fine on [user guide secure connection part](https://github.com/exasol/kafka-connector-extension/blob/main/doc/user_guide/user_guide.md#import-kafka-topic-data-with-ssl-enabled). Then we can merge the pull request.

**Comment by millin** [comment]
Hello @morazow,

I'm working on it, I will make a commit soon.

**Comment by morazow** [comment]
Great, thanks!

Please do not forget to pull last to commits I have added.

**Comment by millin** [comment]
@morazow Please review

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 6]
```diff
@@ -238,6 +247,50 @@ class KafkaConsumerProperties(private val properties: Map[String, String]) exten
...
+   */
+  final def getSASLJaasLocation(): String =
+    get(SASL_JAAS_LOCATION.userPropertyName)
+      .fold(SASL_JAAS_LOCATION.defaultValue)(identity)
+
+  /**
+   * Returns SASL JAAS config file content.
+   */
+  final def getSASLJaasConfig(): String = {
+    val sasl_jaas_location = getSASLJaasLocation()
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:269]:
```suggestion
    val saslJaasLocation = getSASLJaasLocation()
```

Comment (morazow) [reply src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:269]: Please change others, SonarCloud will complain and list them as code smell.

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 7]
```diff
@@ -238,6 +247,50 @@ class KafkaConsumerProperties(private val properties: Map[String, String]) exten
...
+    if (sasl_jaas_location != "") {
+      validateSASLJaasLocationFileExist(sasl_jaas_location)
+      val source = Source.fromFile(sasl_jaas_location)(Codec.UTF8)
+      try source.mkString finally source.close()
+    } else {
+      val username = getString(SASL_USERNAME.userPropertyName)
+      val password = getString(SASL_PASSWORD.userPropertyName)
+      val sasl_module_name: String = if ("PLAIN" == getSASLMechanism()) {
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:277]: This also.

File: src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala [hunk 8]
```diff
@@ -287,6 +344,14 @@ class KafkaConsumerProperties(private val properties: Map[String, String]) exten
   final def mkString(): String =
     mkString(KEY_VALUE_SEPARATOR, PROPERTY_SEPARATOR)
 
+  private[this] def validateSASLJaasLocationFileExist(sasl_jaas_location: String): Unit =
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala:347]: Change parameter to camelCase.

File: doc/user_guide/user_guide.md [hunk 9]
```diff
@@ -535,23 +568,24 @@ These are optional parameters with their default values.
...
+  **'false'**. Use `SECURITY_PROTOCOL=SSL` or `SECURITY_PROTOCOL=SASL_SSL` instead.
 
 * ``SECURITY_PROTOCOL`` - It is the protocol used to communicate with Kafka
   servers. Default value is **PLAINTEXT**.
 
 * ``SSL_KEY_PASSWORD`` - It represents the password of the private key inside
   the keystore file.
 
-* ``SSL_KEYSTORE_PASSWORD`` - It the store password for the keystore file.
+* ``SSL_KEYSTORE_PASSWORD`` - It is the store password for the keystore file.
```
Comment (morazow) [inline doc/user_guide/user_guide.md:588]:
```suggestion
* ``SSL_KEYSTORE_PASSWORD`` - It is the password for the keystore file.
```

File: doc/user_guide/user_guide.md [hunk 10]
```diff
@@ -438,11 +438,44 @@ FROM SCRIPT KAFKA_CONSUMER WITH
...
   SECURITY_PROTOCOL       = 'SSL'
   CONNECTION_NAME         = 'KAFKA_SSL_CONNECTION';
 ```
 
+### Import Kafka Topic Data With Authentication
+
+Create an Exasol named connection object and encode credentials with key-value
+pairs separated by a semicolon (`;`).
+
+Note: Authentication can be used conjunction with encryption.
```
Comment (morazow) [inline doc/user_guide/user_guide.md:450]:
```suggestion
Note: Authentication can be used in conjunction with encryption.
```

**Review by morazow — CHANGES_REQUESTED** [review]
Added minor suggestions

**Review by morazow — APPROVED** [review]

**Comment by morazow** [comment]
Great, looks good!

**Pull request merged**